### DISCUSSION TITLE
Issue with Specifying `server` Parameter in SQLAlchemy URL for MS SQL Server

### DISCUSSION THREAD

**Discussion by nosklo** [sqlalchemy/sqlalchemy#10878] [Usage Questions] [answered]
#### Environment
- SQLAlchemy version: 2.0.25
- Python version: 3.11.1
- OS: Windows 10
- DBAPI: pyodbc

#### Description
I am encountering an issue with SQLAlchemy where I am unable to correctly format a connection string for MS SQL Server without using a DSN (`dsn=MYSERVER`). My requirement is to use `server=MYSERVER` in the connection string.

#### Expected Behavior
I expect to be able to specify `MYSERVER` in the url but the resulting connection string generated by SQLAlchemy defaults to using DSN.
I can connect directly through `pyodbc` with no problems, by using a connection string that uses `SERVER=`, but I would like for a way to translate the host in the url to `SERVER` instead of `DSN`

#### Observed Behavior
Currently, when I specify the server information through the `URL.create` method in SQLAlchemy, it defaults to using DSN (`dsn=MYSERVER`), which is not the desired behavior for my use case.

#### Steps to Reproduce
Here is the relevant part of the code I am using:

```python
import sqlalchemy

connection_info = {
  'SERVER': 'MYSERVER',
  'ENCRYPT': 'no',
  'UID': 'myuser',
  'PWD': 'mypwdhere',
  'TrustServerCertificate': 'yes',
  'driver': '{ODBC Driver 17 for SQL Server}',
}

connection_url = sqlalchemy.engine.URL.create(
    "mssql+pyodbc",
    username=connection_info['UID'],
    password=connection_info['PWD'],
    host=connection_info['SERVER'],
    query={
        "driver": connection_info['driver'],
        "Encrypt": connection_info['ENCRYPT'],
        "TrustServerCertificate": connection_info['TrustServerCertificate']
    }
)
```

I can work around the issue by generating a pyodbc connection string instead, like this:

```python
def fix_mssql_sqlalchemy_url(url):
    if url.get_backend_name() == 'mssql':
        # convert to a pyodbc connection string
        conn_string_data = {
            'driver': '{' + detect_best_mssql_pyodbc_driver() + '}',
            'server': url.host,
            'uid': url.username,
            'pwd': url.password,
        }
        if url.database:
            conn_string_data['database'] = url.database
        if url.port:
            conn_string_data['port'] = url.port
        if url.query:
            conn_string_data.update(url.query)
        conn_string = ';'.join([f'{key}={value}' for key, value in conn_string_data.items()])
        url = sqlalchemy.URL.create('mssql+pyodbc', query={'odbc_connect': conn_string})
    return url

```

When I try to connect to this url, pyodbc gets a connection string that contains `dsn=MYSERVER`; How can I make it go  `server=MYSERVER` instead?

**Comment by gordthompson** [comment, marked as answer]
Try adding `database="master",` to your `URL.create()` call and see if that helps.

**Reply by nosklo** [reply]
That works, thanks! The fix function is a lot shorter now, if there's no database I just add `'master'` to the url.
